IFComp/lib/IFComp/Controller/Vote.pm

```diff
@@ -25,11 +25,15 @@ sub index : Path : Args(2) {
 
     my $current_comp = $c->model('IFCompDB::Comp')->current_comp;
 
-    my $ballot_uri = $c->uri_for_action('/ballot/vote');
-    unless ( $c->req->referer =~ /^$ballot_uri/ ) {
+    my $ballot_uri      = $c->uri_for_action('/ballot');
```

Comment:
Looks like this argument needs to be '/ballot/index'. (Else this results in a "Can't find action for path '/ballot'" exception getting thrown when you try to cast a vote.)